packages/pigeon/lib/java_generator.dart

```diff
@@ -458,7 +458,7 @@ class JavaGenerator extends StructuredGenerator<JavaOptions> {
                       '$returnType $output = channelReply == null ? null : ((Number) channelReply).longValue();');
                 } else {
                   indent.writeln(
-                      '$returnType $output = ($returnType) channelReply;');
+                      '$returnType $output = ${_cast('channelReply', javaType: returnType)};');
```

Comment:
The cast changes in Java and Kotlin fix uncessary casts (`foo as Any?`, `foo as? Any` in Kotlin, `(Object) foo` in Java).